bls/src/test/java/org/dashj/bls/GarbageCollectionTest.java

```diff
@@ -0,0 +1,25 @@
+package org.dashj.bls;
+
+import org.junit.Test;
+
+
+import static org.junit.Assert.assertTrue;
+
+public class GarbageCollectionTest {
+    @Test
+    public void gcTest() {
+        PrivateKey sk = PrivateKey.fromSeedBIP32(Entropy.getRandomSeed(PrivateKey.PRIVATE_KEY_SIZE));
+        G1Element pk = sk.getG1Element();
+
+        BasicSchemeMPL scheme = new BasicSchemeMPL();
+        byte [] message = Entropy.getRandomSeed(32);
+        G2Element sig = scheme.sign(sk, message);
+        assertTrue(scheme.verify(pk, message, sig));
+
+        sk = null;
+        pk = null;
+        sig = null;
+        message = null;
+        System.gc();
```

Comment:
Perhaps we can even check if those objects were collected - e.g. if we hold them in a `WeakReference` and then after `System.gc()` check if the weakref still holds anything.

Reply:
Thanks for the idea.  The main purpose of this test was to check with debug breakpoints if the `finalize()` method of these classes was called.  In finalize, the native shared library is called to `delete` the underlying C++ objects.